server/modules/category/manager/categoryManager.js

```diff
@@ -43,13 +43,13 @@ const insertCategory = async (user, surveyId, category, client = db) =>
     return await assocValidation(Category.assocLevelsArray(levelsDb)(categoryDb))
   })
 
-const insertLevel = async (user, surveyId, categoryUuid, level, client = db) =>
+const insertLevel = async (user, surveyId, level, client = db) =>
   await client.tx(async t => {
-    const levelDb = await CategoryRepository.insertLevel(surveyId, categoryUuid, level, t)
+    const levelDb = await CategoryRepository.insertLevel(surveyId, level, t)
 
     await markSurveyDraft(surveyId, t)
 
-    await ActivityLog.log(user, surveyId, ActivityLog.type.categoryLevelInsert, { categoryUuid, level }, t)
+    await ActivityLog.log(user, surveyId, ActivityLog.type.categoryLevelInsert, { level }, t)
 
```

Comment:
Can these three operations be done concurrently? i.e. with a `Promise.all` or

```
const levelDbPromise = CategoryRepository.insertLevel(surveyId, level, t)
const markPromise = markSurveyDraft(surveyId, t)
const logPromise = ActivityLog.log(user, surveyId, ActivityLog.type.categoryLevelInsert, { level }, t)

await markPromise
await logPromise
return await levelDbPromise
```

Reply:
I'll do it with a Promise.all

Reply:
do you remember that when transactions fails in Promise.all the exception is not catched? I would avoid promise.all